hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/exception/CustomHNSException.java

```diff
@@ -1,24 +1,29 @@
 package ca.bc.gov.hlth.hnsecure.exception;
 
+import ca.bc.gov.hlth.hnsecure.message.ErrorMessage;
+
 /**
  * Custom exception created for HN Error handling 
  * @author pankaj.kathuria
  *
  */
 public class CustomHNSException extends Exception {
 
-	/**
-	 * Added for warning 
-	 */
 	private static final long serialVersionUID = 1L;
+	
+	private ErrorMessage errorMessage;
 
-	public CustomHNSException(String msg) {
-		super(msg);
+	public CustomHNSException(ErrorMessage errorMessage) {
+		super(errorMessage.getErrorMessage());
+		this.errorMessage = errorMessage;
```

Comment:
Switched to use an enum instead of String since it makes for cleaner comparisons.